src/gateways/createArchiveLoader.ts

```diff
@@ -3,30 +3,44 @@ import { ArchiveBook, ArchiveContent, ArchivePage } from '../app/content/types';
 import { stripIdVersion } from '../app/content/utils';
 import { acceptStatus } from '../helpers/fetch';
 
+interface Extras {
+  books: Array<{
+    ident_hash: string
+  }>;
+}
+
 export default (url: string) => {
+  const archiveFetch = <T>(fetchUrl: string) => fetch(fetchUrl)
+    .then(acceptStatus(200, (status, message) => `Error response from archive "${fetchUrl}" ${status}: ${message}`))
+    .then((response) => response.json() as Promise<T>);
+
   const cache = new Map();
-  const loader = memoize((id: string) => fetch(`${url}/${id}`)
-    .then(acceptStatus(200, (status, message) => `Error response from archive "${url}/${id}" ${status}: ${message}`))
-    .then((response) => response.json() as Promise<ArchiveContent>)
+  const contentsLoader = memoize((id: string) => archiveFetch<ArchiveContent>(`${url}/contents/${id}`)
     .then((response) => {
       cache.set(id, response);
       return response;
     })
   );
 
+  const getBookIdsForPage: (pageId: string) => Promise<string[]> =
+    memoize((pageId) => archiveFetch<Extras>(`${url}/extras/${pageId}`).
```

Comment:
:+1: (just noting to self where the code sections are because there are so many tests in the diff 😄 )